backend/v3/storage/database/repository/repository.go

```diff
@@ -65,3 +65,42 @@ func getMany[Target any](ctx context.Context, querier database.Querier, builder
 	}
 	return targets, nil
 }
+
+type updatable interface {
+	PrimaryKeyColumns() []database.Column
+	UpdatedAtColumn() database.Column
+	unqualifiedTableName() string
+}
+
+func updateOne[Target updatable](ctx context.Context, client database.QueryExecutor, target Target, condition database.Condition, changes ...database.Change) (int64, error) {
+	if len(changes) == 0 {
+		return 0, database.ErrNoChanges
+	}
+	if err := checkPKCondition(target, condition); err != nil {
+		return 0, err
+	}
+	if !database.Changes(changes).IsOnColumn(target.UpdatedAtColumn()) {
+		changes = append(changes, database.NewChange(target.UpdatedAtColumn(), database.NullInstruction))
+	}
+	builder := database.NewStatementBuilder(`UPDATE zitadel.` + target.unqualifiedTableName() + ` SET `)
```

Comment:
I would use the `qualifiedTableName` here
```suggestion
	builder := database.NewStatementBuilder(`UPDATE ` + target.qualifiedTableName() + ` SET `)
```